apps/src/lab2/views/dialogs/ShareDialog.tsx

```diff
@@ -179,9 +178,14 @@ const ShareDialog: React.FunctionComponent<{
       : STUDENT_FEEDBACK_LINK;
   });
 
+  // We pull the theme from header redux because the ShareDialog is not wrapped by the lab's
+  // ThemeProvider (the header is in its own tree). We copy the lab theme to the header redux
+  // in ProjectContainer (where we do our other header management).
+  const theme = useAppSelector(state => state.header.theme);
```

Comment:
Could we pass the theme as a prop (via arguments to `shareLab2Project`)?

Reply:
I looked into this as well in https://github.com/code-dot-org/code-dot-org/pull/66019 since we're adding the sharing disabled dialog to `ShareDialog` in the PR. We could pass `theme` to `header.showProjectHeader` and `header.showHeaderForProjectBacked` in `ProjectContainer`.
In `header.js`, these two functions dispatch either `showProjectHeader` or `showProjectBackedHeader` (from `headerRedux`, which just sets `currentHeader` type.
In `ProjectShare` , we could pass theme to `shareLab2Project`. but there doesn't seem to be a way to access`theme`in `ProjectShare` since the lab theme context is not available for this header component.

Reply:
Gotcha - how about storing in `Lab2Registry`? I'm just a little wary of adding theme to header redux since it could get manipulated by other non-lab2 components?